python_modules/libraries/dagster-dg/dagster_dg/templates/COMPONENT_TYPE/COMPONENT_TYPE_NAME_PLACEHOLDER.py.jinja

```diff
@@ -1,21 +1,20 @@
+from dataclasses import dataclass
+
 from dagster import Definitions
 from dagster_components import (
     Component,
     ComponentLoadContext,
     Resolvable,
 )
 
+@dataclass(frozen=True)
```

Comment:
This also updates the template used in scaffolding, please let me know if this is okay.

Reply:
so the scaffolding is worthy of discussion. we specifically wanted to be agnostic to this, as not everyone uses dataclasses

Reply:
Ah understood, for now I will revert that change in this PR.

How do you feel about being opinionated about using data classes in the tutorial? Should we have a tab selection and present both options, or is showing the _dataclass_ sufficient?